scripts/api/highlights.ts

```diff
@@ -19,12 +23,69 @@ function fetchHighlights(): Promise<IHighlightResponse> {
     });
 }
 
+function exportHighlight(id: IArticle['_id'], hasUnsavedChanges: boolean): Promise<void> {
+    const exportRq = () => {
+        return httpRequestJsonLocal<void>({
+            method: 'POST',
+            path: '/generate_highlights',
+            payload: {package: id},
+        }).then(ng.get('authoringWorkspace').edit, (response) => {
+            if (response.internal_error === 403) {
+                forceExportHighlight(id);
+            } else {
+                notify.error(gettext('Error creating highlight.'));
+            }
+        });
+    };
+
+    return hasUnsavedChanges ? ui.confirm(gettext('You have unsaved changes, do you want to continue?'))
+        .then(() => exportRq()) : exportRq();
+}
+
+function forceExportHighlight(id: IArticle['_id']): void {
+    ui.confirm(gettext('There are items locked or not published. Do you want to continue?'))
+        .then(() => {
+            httpRequestJsonLocal({
+                method: 'POST',
+                path: '/generate_highlights',
+                payload: {package: id, export: true},
+            }).then(ng.get('authoringWorkspace').edit, () => {
+                notify.error(gettext('Error creating highlight.'));
+            });
+        });
+}
+
+function previewHighlight(id: IArticle['_id']): Promise<string> {
```

Comment:
It would be good to call it `getDataForHighlightPreview` or `prepareHighlightForPreview`, but not `previewHighlight` - because no actual previewing is happening solely because of this function call.